frontend/appflowy_flutter/lib/workspace/presentation/settings/widgets/settings_language_view.dart

```diff
@@ -106,7 +106,13 @@ class LanguageItem extends StatelessWidget {
             : null,
         onTap: () {
           if (currentLocale != locale) {
-            context.read<AppearanceSettingsCubit>().setLocale(context, locale);
+            try {
+              context
+                  .read<AppearanceSettingsCubit>()
+                  .setLocale(context, locale);
+            } catch (e) {
+              Log.warn('Failed to set locale: $e');
```

Comment:
The error inside appearance.dart has already been caught . No need to catch again here.